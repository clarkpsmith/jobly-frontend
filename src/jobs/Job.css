.Job {
  font-size: x-small;
  padding: 10px;
  margin-bottom: 15px;

  text-align: left;
  box-shadow: 0 0 2px grey;
}

.Job:hover {
  box-shadow: 0 0 6px grey;
}

.Job-btn {
  float: right;
}
