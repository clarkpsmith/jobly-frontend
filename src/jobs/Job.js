import React, { useState, useEffect, useContext } from "react";
import { Button } from "reactstrap";
import addCommas from "../helpers/addCommas";

import UserContext from "../common/UserContext";

import "./Job.css";

const Job = ({ job }) => {
  const [applied, setApplied] = useState("Apply");
  const { apply, currentUser } = useContext(UserContext);

  useEffect(() => {
    async function jobEffect() {
      if (currentUser) {
        const applied = currentUser.applications.find(
          (applicationId) => applicationId === job.id
        );
        setApplied(applied);
      }
    }
    jobEffect();
  }, []);

  function handleClick() {
    apply(currentUser.username, job.id);
    setApplied(job.id);
  }

  const salaryWithCommas = addCommas(job.salary);

  return (
    <div className="Job">
      <h6 className="font-weight-bold"> {job.title}</h6>

      <h6>{job.companyName}</h6>
      <br />
      <p>Salary: {salaryWithCommas ? `$${salaryWithCommas}` : "None"}</p>
      <p>
        Equity: {job.equity === 0 || !job.equity ? "None" : `${job.equity} %`}
        <span>
          {applied ? (
            <Button
              onClick={handleClick}
              size="sm"
              className="Job-btn"
              color="danger"
              disabled
            >
              Applied
            </Button>
          ) : (
            <Button
              onClick={handleClick}
              size="sm"
              className="Job-btn"
              color="danger"
            >
              Apply
            </Button>
          )}
        </span>
      </p>
    </div>
  );
};

export default Job;
