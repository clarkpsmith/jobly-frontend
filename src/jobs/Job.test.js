import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import UserContext from "../common/UserContext";
import Job from "./Job";

const currentUser = {
  applications: [
    200,
    161,
    86,
    50,
    71,
    158,
    36,
    134,
    78,
    184,
    60,
    181,
    89,
    172,
    62,
    82,
    98,
    47,
    183,
    29,
    63,
    53,
    114,
    69,
    7,
    150,
    18,
    95,
    177,
    23,
    123,
    175,
    100,
  ],
  email: "clark.smith79@gmail.com",
  firstName: "Clark",
  isAdmin: false,
  lastName: "Smith",
  username: "testuser",
};

const job = {
  id: 200,
  title: "Accommodation manager",
  salary: 126000,
  equity: null,
  companyHandle: "mejia-scott-ryan",
  companyName: "Mejia, Scott and Ryan",
};

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser }}>
        <Job job={job} />
      </UserContext.Provider>
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser }}>
        <Job job={job} />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
