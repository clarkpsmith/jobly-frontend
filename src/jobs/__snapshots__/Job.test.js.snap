// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Job"
  >
    <h6
      class="font-weight-bold"
    >
       Accommodation manager
    </h6>
    <h6>
      Mejia, Scott and Ryan
    </h6>
    <br />
    <p>
      Salary: $126,000
    </p>
    <p>
      Equity: None
      <span>
        <button
          class="Job-btn btn btn-danger btn-sm disabled"
          disabled=""
          type="button"
        >
          Applied
        </button>
      </span>
    </p>
  </div>
</DocumentFragment>
`;
